source/rest/RestResource.cs

```diff
@@ -4,13 +4,15 @@ namespace com.esendex.sdk.rest
     internal abstract class RestResource
     {
         public abstract string ResourceName { get; }
+        public abstract string ResourceVersion { get; }
 
         public string Content { get; protected set; }
-
         public string ResourcePath { get; protected set; }
+        public string Version { get; protected set; }
 
         public RestResource()
         {
+            Version = ResourceVersion;
```

Comment:
I was just copying the way that ResourcePath is set to the value of ResourceName. I hate it, but didn't really feel like changing how all this works.